Source/SG/SG.cs

```diff
@@ -99,22 +134,66 @@ public static PAENConfiguration GetDefault()
+            string instrumentModel = rfsgHandle.Identity.InstrumentModel;
+
+            if (Regex.IsMatch(instrumentModel, "NI PXIe-5(82.|645R)")) // matches on any baseband instrument without an LO
+                return; // return early since the instrument doesn't have any LOs that can be configured
 
-            if (instrConfig.ShareLOSGToSA)
+            // Set global automatic lo sharing property in rfsgpbl
+            IntPtr rfsgPtr = rfsgHandle.GetInstrumentHandle().DangerousGetHandle();
+            switch (instrConfig.LOSharingMode)
+            {   // StoreAutomaticSGSASharedLO will set LO source for 583x and set LOOutExportConfigureFromRFSA for 584x - happens immediately
+                case LocalOscillatorSharingMode.None:
+                case LocalOscillatorSharingMode.Manual:
+                    // 584x - LOOutExportConfigureFromRFSA set to false
+                    // 583x - LOSource reset
+                    NIRfsgPlayback.StoreAutomaticSGSASharedLO(rfsgPtr, "", RfsgPlaybackAutomaticSGSASharedLO.Disabled);
+                    break;
+                default:
+                    // 584x - LOOutExportConfigureFromRFSA set to true
+                    // 583x - LOSource set to SG_SA_Shared
+                    NIRfsgPlayback.StoreAutomaticSGSASharedLO(rfsgPtr, "", RfsgPlaybackAutomaticSGSASharedLO.Enabled);
+                    break;
+            }
+
+            /// Properties to modify related to LO routing:
+            /// Source
+            /// LOOutEnabled
+            foreach (LocalOscillatorRoutingConfiguration loRoutingConfig in instrConfig.LORoutingConfigurations)
             {
-                rfsgHandle.RF.LocalOscillator.LOOutEnabled = true;
-                rfsgHandle.RF.LocalOscillator.Source = RfsgLocalOscillatorSource.Onboard;
+                switch (instrConfig.LOSharingMode)
+                {
+                    case LocalOscillatorSharingMode.None:
+                        rfsgHandle.RF.LocalOscillator[loRoutingConfig.ChannelName].Source = RfsgLocalOscillatorSource.Onboard;
+                        rfsgHandle.RF.LocalOscillator[loRoutingConfig.ChannelName].LOOutEnabled = false;
+                        break;
+                    case LocalOscillatorSharingMode.Manual:
+                        rfsgHandle.RF.LocalOscillator[loRoutingConfig.ChannelName].Source = RfsgLocalOscillatorSource.FromString(loRoutingConfig.Source);
+                        rfsgHandle.RF.LocalOscillator[loRoutingConfig.ChannelName].LOOutEnabled = loRoutingConfig.ExportEnabled;
+                        break;
+                    default:
+                        if (Regex.IsMatch(instrumentModel, "NI PXIe-584.")) // we do not want to reset LOSource for 583x - would override AutoSGSAShared behavior
+                            rfsgHandle.Utility.ResetAttribute(loRoutingConfig.ChannelName, typeof(RfsgChannelBasedLO).GetProperty("Source"));

```

Comment:
Could we instead just move this whole section (Routing Config) prior to the LO Sharing mode section? That way, we would not reset the LO source inappropriately.

Reply:
short answer is no. there are other things that StoreAutomaticSGSASharedLO is doing that won't make that an easy change